src/index.js

```diff
@@ -0,0 +1,324 @@
+const OTPAuth = require('otpauth');
+import CryptoJS from "crypto-js";
+import browser from 'browser-detect';
+import crypto from 'crypto';
+import eccrypto from 'eccrypto';
+import BeetClientDB from './lib/BeetClientDB';
+import "isomorphic-fetch";
+
+
+
+
+
+
+
+
+
+class Beet {
+
+    constructor() {
+        this.host = 'wss://local.get-beet.io:60556';
+        this.connected = false;
+        this.authenticated = false;
+        this.linked = false;
+        this.initialised = false;
+        this.socket = null;
+        this.appName = null;
+        this.otp = null;
+        this.openRequests = [];
+        this.origin = null;
+        this.socket = null;
+    }
+    reset() {
+        this.connected = false;
+        this.authenticated = false;
+        this.linked = false;        
+        this.socket = null;        
+        this.otp = null;
+        this.openRequests = [];        
+        this.socket = null;
+    }
+    async init(app) {
+        if (this.initialised === false) {
+            this.appName = app;
+            this.origin = app;
+            if (typeof location !== 'undefined') {
+                if (location.hasOwnProperty('hostname') && location.hostname.length && location.hostname !== 'localhost') {
+                    this.origin = location.hostname;
+                }
+            }
+        }
+        this.detected = browser();
+        this.keyhash = CryptoJS.SHA256(this.detected.name + ' ' + this.origin + ' ' + this.appName).toString();
+        let appstore;
+        appstore = await BeetClientDB.apps.where("keyhash").equals(this.keyhash).toArray();
+        this.initialised = true;
+        return appstore;
+    }
+    ping() {
+        let ping;
+        return new Promise(async (resolve, reject) => {
+            try {
+                ping = new WebSocket(this.host);
+                ping.onopen = function (evt) {
+                    ping.send('{ "type" : "version"}');
+                }
+                ping.onmessage = function (evt) {
+                    let msg = JSON.parse(evt.data);
+                    if (msg.type == "version") {
+                        resolve(msg.result);
+                    } else {
+                        reject(false);
+                    }
+                    ping.close();
+                }
+            } catch (e) {
+                reject(false);
+            }
+        });
+    }
+    isInstalled() {
+        return new Promise(resolve => {
+            setTimeout(() => {
+                resolve(false);
+            }, 3000);
```

Comment:
3 second is too long. It will be resolving on localhost, I suggest something < 500ms